src/compilation/options.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{
+    compilation::{compile_validators, context::CompilationContext, JSONSchema, DEFAULT_SCOPE},
+    error::CompilationError,
+    resolver::Resolver,
+    schemas,
+};
+use serde_json::Value;
+use std::borrow::Cow;
+
+#[derive(Clone, Debug, Default)]
+pub struct CompilationOptions {
+    pub(crate) draft: Option<schemas::Draft>,
+}
+
+impl CompilationOptions {
+    pub(crate) fn draft(&self) -> schemas::Draft {
```

Comment:
maybe it could be e.g. `draft_or_default`? so `draft()` could be used as a setter. I am not sure which naming scheme will be the most intuitive